misk-zookeeper/src/test/kotlin/misk/clustering/zookeeper/ZkLeaseTest.kt

```diff
@@ -212,6 +217,48 @@ internal class ZkLeaseTest {
     assertThat(lease.checkHeld()).isFalse() // still not held since its closed
   }
 
+  @Test fun listenerNotifications() {
+    val listenerMock = Mockito.mock<Lease.StateChangeListener>()
```

Comment:
I don't usually mocks, but this seemed appropriate since I want to verify exactly how many times the listener gets called.

Reply:
Could also just increment an `AtomicInteger` and check the result...